src/aliceVision/sfm/pipeline/sequential/ReconstructionEngine_sequentialSfM.cpp

```diff
@@ -1702,6 +1696,16 @@ void ReconstructionEngine_sequentialSfM::triangulateMultiViews_LORANSAC(SfMData&
         scene.structure[trackId] = landmark;
       }      
     }
+    else
+    {
+      if (scene.structure.find(trackId) != scene.structure.end()) 
```

Comment:
You cannot call `scene.structure.find(trackId)` without locking as it is modified inside the loop.
```
else if(observations.size() > 2) // with only 2 observations it can only be a new landmark (so nothing to remove)
{
    #pragma omp critical
    if (scene.structure.find(trackId) != scene.structure.end())
    {
        scene.structure.erase(trackId);
    }
}
```

Reply:
find() is a reading function... I put #pragma on writing functions only. Is it wrong?

Reply:
It may crash if you read something while another thread is writing.

Reply:
Excepted if the writing parts are well locked.